lib/convert.js

```diff
@@ -3,17 +3,33 @@
 // ParameterKey=string,ParameterValue=string,UsePreviousValue=boolean
 const opt_params = (param) => {
   const retrived_param = {};
-  const pairs = param.split(',').map((pair) => pair.split('='));
+  const param_arr = param.match(/(\s*([^,]+)".*?"|\s*([^,]+)[^",\s]+)(?=\s*,|\s*$)/g);
+  /* will match:
+      (\s*([^,]+)     Any string
+      AND
+      (
+          ".*?"       double quotes + anything but double quotes + double quotes
+          |           OR
+          [^",\s]+    1 or more characters excl. double quotes, comma or spaces of any kind
+      )
+      (?=             FOLLOWED BY
+          \s*,        0 or more empty spaces and a comma
+          |           OR
+          \s*$        0 or more empty spaces and nothing else (end of string)
+      )
+  */
+  
+  param_arr.forEach((pair) => {
+    const split_pair = pair.split(/=(.+)?/);
 
-  pairs.forEach((pair) => {
-    if (pair[0] === 'ParameterKey') {
-      retrived_param.ParameterKey = pair[1];
+    if (split_pair[0] === 'ParameterKey') {
+      retrived_param.ParameterKey = split_pair[1];
     } 
-    else if (pair[0] === 'ParameterValue') {
-      retrived_param.ParameterValue = pair[1];
+    else if (split_pair[0] === 'ParameterValue') {
+      retrived_param.ParameterValue = split_pair[1].replace(/^"(.*)"$/, '$1');
```

Comment:
`... .replace('"', '')?`

Reply:
I believe `.replace('"', '')` is removing all the double quotes whereas `replace(/^"(.*)"$/, '$1')` is removing only the first and last `"`.

Reply:
`.replace(/(^"|"$)/g, '')`?